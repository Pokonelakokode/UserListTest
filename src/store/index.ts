import {configureStore, combineReducers} from "@reduxjs/toolkit";
import thunk from "redux-thunk";
import {pageStateReducer} from "./reducers/pageState";

export const rootReducer = combineReducers({
    pageState: pageStateReducer,
});
export type RootState = ReturnType<typeof rootReducer>;

export const store = configureStore<RootState>({
    devTools: process.env.NODE_ENV === "production" ? false : {trace: true, traceLimit: 25},
    middleware: [thunk],
    reducer: rootReducer,
});
