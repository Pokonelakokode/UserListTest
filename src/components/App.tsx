import * as React from "react";
import {useDispatch, useSelector} from "react-redux";
import { bindActionCreators } from "redux";
import {router} from "../router";
import {mainSelector} from "../store";
import {navigate} from "../store/actions/hashRouter";
import {Pages} from "../store/reducers/pageState";
import {getHash} from "../utils";

const App: React.FC = (props) => {
    const {pageState: {page}} = useSelector(mainSelector);
    const actions = bindActionCreators({navigate }, useDispatch());
    React.useEffect(() => {
        const fn = () => actions.navigate({
            page: getHash().page || Pages.ALL,
            userId: getHash().userId || null,
         });
        window.addEventListener("popstate", fn);
        return () => window.removeEventListener("popstate", fn);
    }, []);
    return router(page);
};

export default App;
