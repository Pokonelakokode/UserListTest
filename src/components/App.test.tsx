import { create } from "react-test-renderer";
import {Provider} from "react-redux";
import {store} from "../store";
import React from "react";
import App from "./App";

describe("App component", () => {
    it("should be able to render", () => {
        const component = create(
            <Provider store={store}>
                <App/>
            </Provider>,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});