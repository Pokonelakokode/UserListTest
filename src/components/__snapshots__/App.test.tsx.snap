// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should be able to render 1`] = `
<div
  id="UsersList"
>
  <h2>
    Users
  </h2>
  <div
    className="controls"
  >
    <input
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
    <div
      className="pager"
    >
      <div
        className="arrow"
        onClick={[Function]}
      >
        ⇐
      </div>
      <div>
        Users: 
        <b>
          0
        </b>
        <br />
        Page: 
        <b>
          1
          /
          0
        </b>
      </div>
      <div
        className="arrow"
        onClick={[Function]}
      >
        ⇒
      </div>
    </div>
  </div>
  <div
    className="users-list"
  />
  <button
    className="new-user"
    onClick={[Function]}
  >
    New User
  </button>
</div>
`;
